generation/new_client_hermetic_build/new-client.py

```diff
@@ -0,0 +1,330 @@
+    __add_item_if_set(new_library, "issue_tracker", issue_tracker)
+    __add_item_if_set(new_library, "extra_versioned_modules", extra_versioned_modules)
+
+    config["libraries"].append(new_library)
+    config["libraries"] = sorted(config["libraries"], key=__compute_library_name)
+
+    with open(path_to_yaml, "w") as file_stream:
+        yaml.dump(config, file_stream)
+
+
+def __add_item_if_set(target: dict, key: str, value: any) -> None:
+    if value is not None:
+        target[key] = value
+
+
+def __compute_library_name(library: dict) -> str:
+    if "library_name" in library:
+        return f'java-{library["library_name"]}'
+    return f'java-{library["api_shortname"]}'
+
+
+def __get_proto_paths(proto_path: str, committish: str) -> List[str]:
+    version_re = re.compile(r"v\d[\w\d]*")
+    is_library_version = lambda p: version_re.match(p.split("/")[-1]) is not None
+
+    if is_library_version(proto_path):
+        raise ValueError(
+            "Versioned proto_paths are not supported, please provide the root"
+            " proto_path. For example `google/datastore` instead of `google/datastore/v1`."
+            " If all you need is to add a new version to an existing library,"
+            " you need to manually add the new version to `generation_config.yaml`"
+        )
+
+    # if the proto path is not versioned, we must search for all versions
+    # in googleapis
+    if os.path.isdir("./googleapis"):
+        googleapis_repo = Repo("./googleapis")
+    else:
+        googleapis_repo = Repo.clone_from(
+            "https://github.com/googleapis/googleapis", "./googleapis"
+        )
+    googleapis_repo.git.checkout(committish)
+    subdirs = os.walk(f"./googleapis/{proto_path}")
+    proto_paths = []
+    for subdir_path, _, _ in subdirs:
+        if is_library_version(subdir_path) and __is_releasable_version(subdir_path):
+            print(subdir_path)
+            proto_paths.append(subdir_path.replace("./googleapis/", ""))
+    shutil.rmtree("./googleapis")
+    return proto_paths
+
+
+def __is_releasable_version(versioned_proto_path):
+    """true if the library definition found in `versioned_proto_path`:
+    - has a BUILD file
+    - the BUILD file has a `java_gapic_assembly_gradle_pkg` rule
+    """
+    build_files = [
+        file for file in os.listdir(versioned_proto_path) if "BUILD.bazel" in file
+    ]
+    if len(build_files) == 0:
+        print(f"BUILD file not found in {versioned_proto_path}")
+        return False
+
+    build_file = versioned_proto_path + "/" + build_files[0]
+    with open(build_file) as build_file_stream:
+        build_file_contents = build_file_stream.read()
+        gradle_pkg_matcher = re.compile(r"java_gapic_assembly_gradle_pkg")
+        matched_lines = len(
+            [
+                line
+                for line in build_file_contents.split("\n")
+                if gradle_pkg_matcher.search(line) != None
+            ]
+        )
+        return matched_lines == 2
```

Comment:
Why there're two matching lines?

Reply:
One for the [import](https://github.com/googleapis/googleapis/blob/2c41d643ab0e03543dbd92a6ef93da28733853c9/google/ai/generativelanguage/v1/BUILD.bazel#L53) and one for the [build rule](https://github.com/googleapis/googleapis/blob/2c41d643ab0e03543dbd92a6ef93da28733853c9/google/ai/generativelanguage/v1/BUILD.bazel#L100-L110)

I added a comment above this line explaining this